src/libs/elektra/keyset.c

```diff
@@ -919,12 +926,26 @@ ssize_t ksAppendKey (KeySet * ks, Key * toAppend)
 		++ks->size;
 		if (ks->size >= ks->alloc)
 		{
-			if (ksResize (ks, ks->alloc * 2 - 1) == -1)
+			size_t newSize = ks->alloc * 2;
+
+			// If array was not allocated before
+			if (newSize == 0)
+				newSize = KEYSET_SIZE - 1;
+			else
+				newSize = newSize - 1;
+
+			if (ksResize (ks, newSize) == -1)
 			{
 				keyDel (toAppend);
 				--ks->size;
 				return -1;
 			}
+
+			// If array was allocated in ksResize, size is 0
+			if (ks->size == 0)
```

Comment:
As far as I can tell, `ksResize` doesn't modify `ks->size` and `ks->size` was incremented in line 919, so it shouldn't be 0.

Reply:
This is in the case when the array wasn't yet allocated. Then `ksResize` will allocate the array for the first time.

https://github.com/ElektraInitiative/libelektra/blob/170cf57faba58979a79e0c2bf03ddd9545aa21e4/src/libs/elektra/keyset.c#L2679-L2682

And then the increment from earlier is overwritten, so we have to do it again.

The comment (trying) to point this out is not very clear. I'll improve that.